services/galleryService/galleryService.go

```diff
@@ -0,0 +1,105 @@
+package galleryService
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type JSONPhotosets struct {
+	Photosets struct {
+		Photoset []struct {
+			ID      string
+			Primary string
+			Secret  string
+			Server  string
+			Farm    int
+			Title   struct {
+				Content string
+			}
+			Description struct {
+				Content string
+			}
+		}
+	}
+}
+
+type Album struct {
+	Title       string
+	Description string
+	Link        string
+	ImageURL    string
+}
+
+var jsonBytes []byte
+var photos JSONPhotosets
+
+func init() {
+
+	if os.Getenv("FLICKR_API_KEY") == "" {
+		log.Println("Environment variable FLICKR_API_KEY is not assigned")
+		return
+	}
+
+	if os.Getenv("FLICKR_USER_ID") == "" {
+		log.Println("Environment variable FLICKR_USER_ID is not assigned")
+		return
+	}
+
+	apiKey := os.Getenv("FLICKR_API_KEY")
+	userID := os.Getenv("FLICKR_USER_ID")
+	hc := http.Client{}
+	apiURL := "https://api.flickr.com/services/rest/"
+	form := url.Values{}
+	form.Add("method", "flickr.photosets.getList")
+	form.Add("format", "json")
+	form.Add("api_key", apiKey)
+	form.Add("user_id", userID)
+
+	req, err := http.NewRequest("POST", apiURL, strings.NewReader(form.Encode()))
+	if err != nil {
+		panic(err)
+	}
+	req.PostForm = form
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	res, err := hc.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer res.Body.Close()
+
+	jsonBytes, err = ioutil.ReadAll(res.Body)
+	if err != nil {
+		panic(err)
+	}
+	jsonBytes = jsonBytes[14 : len(jsonBytes)-1]
+	jsonBytes = bytes.Replace(jsonBytes, []byte("_"), []byte(""), -1)
```

Comment:
do you need me to explain how the function works? If so, can't I just add a link to the online reference?